src/messaging/ExchangeContext.cpp

```diff
@@ -79,6 +83,24 @@ void ExchangeContext::SetResponseTimeout(Timeout timeout)
     mResponseTimeout = timeout;
 }
 
+#if CONFIG_DEVICE_LAYER && CHIP_DEVICE_CONFIG_ENABLE_SED
+void ExchangeContext::UpdateSEDPollingMode(Transport::Type transportType)
+{
+    if (transportType != Transport::Type::kBle)
+    {
+        if (!IsResponseExpected() && !IsSendExpected() && (mExchangeMgr->GetNumActiveExchanges() == 1) &&
```

Comment:
If `!IsResponseExpected()` && `!IsSendExpected()`, then why is this exchange even in existence?

Reply:
That is because `UpdateSEDPollingMode` method is called few lines before closing the exchange: https://github.com/project-chip/connectedhomeip/pull/11314/files/2691349990d9ac3a330fd02e25bf4b6b98e77bf2#diff-4530316862f734c5993cdcd2f679fa3dc31f3c8b8bfd87a5bb7b25389a3b802dR499